test/support.ts

```diff
@@ -441,6 +443,18 @@ export function expectsql(
     }
   }
 
+  if (typeof query === 'function') {
+    try {
+      query = query();
+    } catch (error: unknown) {
+      if (!(error instanceof Error)) {
+        throw new TypeError('expectsql: function threw something that is not an instance of Error.');
```

Comment:
This doesn't seems helpful

Reply:
I think catch always throws an error, just the instance is different, so maybe we can provide the error based on the instance.